File: DSA/Stack/lab1.c
//LAB 1: implementation of Stack as ADT
#include <stdio.h>
#include <stdlib.h>
#define MAX 5
#define true 1
#define false 0
struct stack
{
 int TOP;
 int data[MAX];
};
int isFull(struct stack s)
{
    return (s.TOP == MAX - 1)? true : false;
}
int isEmpty(struct stack s)
{
    return (s.TOP == -1)? true : false;
}
void push(struct stack *s,int element)
{
    if(isFull(*s))   // sending value of s to the function as *s is pointer.
    {
        printf("Stack Overflow.\n");
    }
    else
    {
        s->TOP +=1;
        s->data[s->TOP] = element;
        printf("Success.\n");
    }
}
int pop(struct stack *s)
{
    if(isEmpty(*s))
    {
        printf("Stack Underflow.\n");
        return -1;
    }
    else 
        return s->data[s->TOP--];

}
int main()
{
    int choice,element;
    struct stack any = {-1}; //or intialize any and any.TOP = -1;
    do
    {
        printf("1.PUSH\n2.POP\n3.EXIT\n");
        printf("Choice? ");
        scanf("%d",&choice);
        switch(choice)
        {
            case 1:
                printf("Enter element\n");
                scanf("%d",&element);
                push(&any,element);
                break;
            case 2:
                printf("%d was popped\n",pop(&any));
                break;
            case 3:
                printf("Bye Bye!\n");
                break;
        }
    } while (choice != 3);
   return 0; 
}

File: DSA/Stack/lab1.py
def push(stack, element):
    stack.append(element)
    print("Success")
def pop(stack):
    if len(stack)==0:
        return "Stack Underflow."
    else:
        data = stack.pop()
        return data

flag = True
stack = list()
while(flag):
    print("1. PUSH")
    print("2.POP")
    print("3.EXIT")
    choice = int(input("Enter you choice: "))
    if choice == 1:
        element = input("Enter element")
        push(stack,element)
    elif choice == 2 :
        element = pop(stack)
        print("Popped Value=",element)
    else:
        flag = False
    